File: backend/app/schemas/astro.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, Dict, List, Union

class BirthDataRequest(BaseModel):
    name: str
    birth_date: str  # YYYY-MM-DD
    birth_time: str  # HH:MM
    birth_place: str
    latitude: float
    longitude: float
    timezone: str

class PlanetPosition(BaseModel):
    name: str
    sign: str
    degree: float
    house: int
    retrograde: bool = False

class BirthChart(BaseModel):
    sun_sign: str
    moon_sign: str
    rising_sign: str
    planets: List[PlanetPosition]
    houses: Dict[str, str]
    aspects: List[Dict[str, Union[str, float]]]

class AstroResponse(BaseModel):
    birth_chart: BirthChart
    raw_data: Dict

File: frontend/src/types/astro.ts
export interface BirthData {
  name: string;
  birth_date: string;
  birth_time: string;
  birth_place: string;
  latitude: number;
  longitude: number;
  timezone: string;
}

export interface PlanetPosition {
  name: string;
  sign: string;
  degree: number;
  house: number;
  retrograde: boolean;
}

export interface BirthChart {
  sun_sign: string;
  moon_sign: string;
  rising_sign: string;
  planets: PlanetPosition[];
  houses: Record<string, string>;
  aspects: Array<Record<string, string>>;
}

export interface AstroResponse {
  birth_chart: BirthChart;
  raw_data: Record<string, any>;
}